lib/credentials/ecs_credentials.js

```diff
@@ -119,13 +139,15 @@ AWS.ECSCredentials = AWS.util.inherit(AWS.Credentials, {
     this.request(relativeUri, function(err, data) {
       if (!err) {
         try {
-          var creds = JSON.parse(data);
-          if (self.credsFormatIsValid(creds)) {
-            self.expired = false;
-            self.accessKeyId = creds.AccessKeyId;
-            self.secretAccessKey = creds.SecretAccessKey;
-            self.sessionToken = creds.Token;
-            self.expireTime = new Date(creds.Expiration);
+          data = JSON.parse(data);
+          if (self.credsFormatIsValid(data)) {
+            var creds = {
```

Comment:
I realized I misunderstood your question. It is indeed necessary to assign these fields to `self`, I've just simply moved them up into the `callbacks` function, and I'm passing this `creds` object containing all of the needed fields into `callbacks`. On line 118 above (`if (!err) AWS.util.update(call.provider, creds);`) is where these fields are getting assigned to the credential provider instance. The reason I can no longer use `self` is because if there are multiple callbacks in the `refreshQueue` coming from multiple instances of the ECS credential provider, only the first `refresh()` call will actually call `this.request()`, so `self` will only refer to that one instance of the credential provider. The `call.provider` in the `refreshQueue` is a reference to the instance of the credential provider where the corresponding callback in the `refreshQueue` is from. Before, assigning the fields directly to `self` was not a problem because every single `refresh()` call would invoke `'this.request()` and the right instance of the credential provider would be referenced.